File: java/no-package/iteration/Factorial.java
import java.util.*;

public class Factorial {
    public static void main(String[] args) {
        int n, f = 1;
        Scanner scanner = new Scanner(System.in);
        System.out.printf("Enter a number: ");
        n = scanner.nextInt();
        for (int i = n; i >= 1; i--) {
            f = f * i;
        }
        System.out.printf("The factorial of %d is %d.\n", n, f);
        scanner.close();
    }
}

File: c/iteration/factorial.c
#include <stdio.h>
int main()
{
    int n, f = 1, i;
    printf("Enter a number: ");
    scanf("%d", &n);
    for (i = n; i >= 1; i--)
    {
        f = f * i;
    }
    printf("The factorial of %d is %d.\n", n, f);
    return 0;
}
